File: Static Factories/Factory.java
public static class Coordinates {

    private final int x;
    private final int y;

    static private final Coordinates maxCoordinates = new Coordinates(100, 100);

    private Coordinates(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public static Coordinates getCoordinates(int x, int y) {
        return new Coordinates(x, y);
    }

    public static Coordinates doubleCoordinates(int x, int y) {
        return new Coordinates(x * 2, y * 2);
    }

    public static Coordinates maximumCoordinates() {
        return maxCoordinates;
    }

    public String toString() {
        return x + " " + y;
    }
} 

File: Static Factories/Factory.scala

class Coordinates private (val x: Int, val y: Int) {
    override def toString = s"$x - $y"
}

object Coordinates {
    private final val maximumCoordinates = new Coordinates(100, 100)

    def maximum = maximumCoordinates
    def getCoordinates(x: Int, y: Int) = new Coordinates(x, y)
    def doubledCoordinates(x: Int, y: Int) = new Coordinates(x * 2, y * 2)
}


object Main extends App {

    val coord1 = Coordinates.doubledCoordinates(4, 10)
    val coord2 = Coordinates.getCoordinates(10, 4)
    
}